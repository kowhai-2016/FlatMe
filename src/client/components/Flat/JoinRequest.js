import React, { PropTypes } from 'react'
import { Link } from 'react-router'

const JoinRequest = props => {
  const request = props.request
  const user = request.user
  const { id: userId, firstName, lastName } = user
  const accept = () => props.acceptJoinRequest(request.id)
  const ignore = () => props.ignoreJoinRequest(request.id)
  return (
    <li>
      <Link to={`/user/${userId}`}>
        {firstName} {lastName}
      </Link>
      <div>
        <button
          className='btn btn-success glyphicon glyphicon-ok'
          onClick={accept}
          >
          Accept
        </button>
        <button
          className='btn btn-danger glyphicon glyphicon-remove'
          onClick={ignore}
          >
          Ignore
        </button>
      </div>
    </li>
  )
}

JoinRequest.propTypes = {
  acceptJoinRequest: PropTypes.func.isRequired,
  ignoreJoinRequest: PropTypes.func.isRequired,
  request: PropTypes.shape({
    id: PropTypes.number.isRequired,
    user: {
      id: PropTypes.number.isRequired,
      firstName: PropTypes.string.isRequired,
      lastName: PropTypes.string.isRequired
    }.isRequired
  }).isRequired
}

export default JoinRequest
