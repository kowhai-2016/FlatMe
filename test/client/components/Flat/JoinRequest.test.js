/* eslint-env jest */

import React from 'react'
import renderer from 'react-test-renderer'

import JoinRequest from '../../../../src/client/components/Flat/JoinRequest'

test('JoinRequest renders correctly', () => {
  const acceptJoinRequest = () => {}
  const request = {
    id: 1,
    user: {
      id: 1,
      firstName: 'First',
      lastName: 'Last'
    }
  }
  const ignoreJoinRequest = () => {}
  const tree = renderer.create(
    <JoinRequest
      acceptJoinRequest={acceptJoinRequest}
      request={request}
      ignoreJoinRequest={ignoreJoinRequest}
      />
  ).toJSON()
  expect(tree).toMatchSnapshot()
})
