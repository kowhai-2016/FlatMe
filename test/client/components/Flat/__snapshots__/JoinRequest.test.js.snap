exports[`test JoinRequest renders correctly 1`] = `
<li>
  <a
    onClick={[Function]}
    style={Object {}}>
    First
     
    Last
  </a>
  <div>
    <button
      className="btn btn-success glyphicon glyphicon-ok"
      onClick={[Function]}>
      Accept
    </button>
    <button
      className="btn btn-danger glyphicon glyphicon-remove"
      onClick={[Function]}>
      Ignore
    </button>
  </div>
</li>
`;
